### DISCUSSION TITLE
How to upgrade ZeroTier controller version?

### DISCUSSION THREAD

**Discussion by timothyl13241** [dec0dOS/zero-ui#28] [Q&A] [answered]
As the most recent version of ZeroTier is 1.8.1, I was wondering if there is any way for me to manually upgrade from 1.6.5, or do I have to wait for the Docker Hub repository to be updated?

Thank you!

**Comment by dec0dOS** [comment, marked as answer]
Hello, @timothyle97!

If you need another version, all you need to do is to set the tag in the `zero-ui/docker/zerotier/Dockerfile` and rebuild the image.
Please refer to https://hub.docker.com/r/zyclonite/zerotier/tags for tag list.


ZeroUI is tested with the 1.6.5, so it may require some time to check the 1.8.1. The controller version does not matter that much, so I'll recommend you to stay with the current one.

**Reply by timothyl13241** [reply]
Hi @dec0dOS,

Thank you for the quick reply! I'm a little new to docker so I'm not sure what you mean by rebuilding the image. I checked the Dockerfile and it is already using the latest tag for the zyclonite/zerotier repository, so I'm not sure why it is still installing 1.6.5.

Does this have to do with the docker-compose.yml specifying the dec0dos/zerotier-controller image?

**Reply by dec0dOS** [reply]
The `dec0dos/zerotier-controller` image was built when the `latest` was 1.6.5, so it stuck to this one.
I didn't pin the version to always test the ZeroUI with the latest ZeroTier Controller one. Maybe I should pin it to provide a better reproducibility when building ZeroUI from the source.

Please follow these steps to upgrade the ZeroTier controller:
1. Clone the repo
`git clone https://github.com/dec0dOS/zero-ui/ && cd zero-ui`
2. Build the image
`docker build -f docker/zerotier/Dockerfile -t zerotier-controller`
3. Change `dec0dos/zerotier-controller` to `zerotier-controller` in docker-compose.yml

**Reply by timothyl13241** [reply]
Thank you so much. This worked and so far 1.8.1 is working fine with ZeroUI.
